homeassistant/components/yolink/__init__.py

```diff
@@ -93,33 +173,14 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     yolink_http_client = YoLinkClient(auth_mgr)
     yolink_mqtt_client = MqttClient(auth_mgr)
-    coordinator = YoLinkCoordinator(hass)
-
+    coordinator = YoLinkCoordinator(hass, yolink_http_client, yolink_mqtt_client)
+    await coordinator.init_coordinator()
```

Comment:
We also need to do a first refresh with `coordinator.async_config_entry_first_refresh`.